### ISSUE TITLE
Input_nodes Type have to be torch.BoolTensor

### ISSUE THREAD

**Issue by RobinBuchta** [pyg-team/pytorch_geometric#7855]
### 📚 Describe the documentation issue

I got two different results for torch.LongTensor and torch.BoolTensor. The result with torch.BoolTensor seems to be correct and with the datatype torch.LongTensor I get very strange results. 

My function looks something like this:
```
def create_neighborloader(data):
    # create a boolean tensor size of data['flow'].x with 1 for all nodes as train_mask
    data['flow'].train_mask = torch.ones(data['flow'].x.size(0), dtype=torch.bool) # vs dtype=torch.long
 
    loader = NeighborLoader(data, shuffle=False, num_neighbors={key: [-1] * 2 for key in data.edge_types}, directed= True, batch_size=SUBGRAPH_SIZE*BATCH_SIZE, input_nodes=['flow', data['flow'].train_mask])
  
    return loader
```

 
 

- Corrected doc entry:  

 input_nodes (torch.Tensor or str or Tuple[str, torch.Tensor]): The
            indices of nodes for which neighbors are sampled to create
            mini-batches.
            **Needs to be :obj:`torch.BoolTensor`.**
            If set to :obj:`None`, all nodes will be considered.
            In heterogeneous graphs, needs to be passed as a tuple that holds
            the node type and node indices. (default: :obj:`None`)

### Suggest a potential alternative/fix

...
input_nodes (torch.Tensor or str or Tuple[str, torch.Tensor]): The
            indices of nodes for which neighbors are sampled to create
            mini-batches.
            Needs to be :obj:`torch.BoolTensor`.
            If set to :obj:`None`, all nodes will be considered.
            In heterogeneous graphs, needs to be passed as a tuple that holds
            the node type and node indices. (default: :obj:`None`)
...

**Comment by rusty1s** [comment]
Can you clarify the issue? You should be able to pass both boolean masks and an index `LongTensor` to `input_nodes`, depending on whether you have your training split given via a mask or indices.

**Comment by RobinBuchta** [comment]
Ah, sorry. Now I understand. It took me a lot of time because I had my mask as a long tensor. But sure, now it makes click: mask or indices and for it the different types.
